zio-http-testkit/src/test/scala/zio/http/TestClientSpec.scala

```diff
@@ -56,43 +55,30 @@ object TestClientSpec extends ZIOSpecDefault {
       ),
       suite("socket ops")(
         test("happy path") {
-          val messageUnwrapper: Http[Any, Nothing, WebSocketChannelEvent, (WebSocketChannel, String)] =
-            Http.collect[WebSocketChannelEvent] {
-              case ChannelEvent(channel, ChannelRead(WebSocketFrame.Text(message))) =>
-                (channel, message)
-            }
-
-          val greetingToClient                                                       = "Hi Client"
-          val messageSocketClient: Http[Any, Throwable, WebSocketChannelEvent, Unit] = messageUnwrapper >>>
-            Handler
-              .fromFunctionZIO[(WebSocketChannel, String)] { case (ch, `greetingToClient`) =>
-                ch.writeAndFlush(WebSocketFrame.text("Hi Server"), await = true)
-              }
-
-          val channelSocketClient: Http[Any, Throwable, WebSocketChannelEvent, Unit] =
-            Http.empty
+          val socketClient: Http[Any, Throwable, WebSocketChannel, Unit] =
+            Http.webSocket { channel =>
+              channel.receive.flatMap {
+                case ChannelEvent.Read(WebSocketFrame.Text("Hi Client")) =>
+                  channel.send(Read(WebSocketFrame.text("Hi Server")))
 
-          val messageSocketServer: Http[Any, Throwable, WebSocketChannelEvent, Unit] = messageUnwrapper >>>
-            Handler
-              .fromFunctionZIO[(WebSocketChannel, String)] { case (ch, "Hi Server") =>
-                ch.close()
-              }
-
-          val channelSocketServer: Http[Any, Throwable, WebSocketChannelEvent, Unit] =
-            Http.collectZIO[WebSocketChannelEvent] {
-              case ChannelEvent(ch, UserEventTriggered(UserEvent.HandshakeComplete)) =>
-                ch.writeAndFlush(WebSocketFrame.text(greetingToClient))
+                case _ =>
+                  ZIO.unit
+              }.forever
             }
 
-          val httpSocketClient: Http[Any, Throwable, WebSocketChannelEvent, Unit] =
-            messageSocketClient ++ channelSocketClient
+          val socketServer: Http[Any, Throwable, WebSocketChannel, Unit] =
+            Http.webSocket { channel =>
+              channel.receive.flatMap {
+                case ChannelEvent.Read(WebSocketFrame.Text("Hi Server")) =>
+                  channel.send(Read(WebSocketFrame.text("Hi Client")))
 
-          val httpSocketServer: Http[Any, Throwable, WebSocketChannelEvent, Unit] =
-            messageSocketServer ++ channelSocketServer
+                case _ => ZIO.unit
+              }.forever
+            }
 
           for {
-            _        <- TestClient.installSocketApp(httpSocketServer)
-            response <- ZIO.serviceWithZIO[Client](_.socket(pathSuffix = "")(httpSocketClient.toSocketApp))
+            _        <- TestClient.installSocketApp(socketServer)
+            response <- ZIO.serviceWithZIO[Client](_.socket(pathSuffix = "")(socketClient.toSocketApp))
```

Comment:
`toSocketApp` can be deleted now, can't it?

Reply:
`socketClient` here is defined as an `Http` so we need to go from an `Http` to a `Handler`, which normally requires specifying a default handler.